### ISSUE TITLE
Error parsing JWT with non-latin characters

### ISSUE THREAD

**Issue by JohnHoward9** [open-policy-agent/opa#5575]
## Short description 
We need to validate and parse the incoming JWT.  But, this is failing when parsing JWTs with umlaut character

Examples:
* OPA version is 0.44

## Steps To Reproduce
OPA parsing JWT token, also an issue if parsing message body with non-latin characters.  We have worked round the latter issue by encoding and decoding those characters.

## Expected behavior
OPA handles non-latin characters

**Comment by srenatus** [comment]
Thanks for the report. Can you please elaborate "fails" here? What's your call, what happens that should not happen? It would help to know the error message and example inputs that lead to it.

**Comment by charlieegan3** [comment]
This is an example playground where the user's name is `Älex` https://play.openpolicyagent.org/p/60PWp3rfEA. OPA appears to have the correct behaviour in this instance.

**Comment by anderseknert** [comment]
Works here too. Like @srenatus said, could you please provide an example for us to reproduce this with?
```
package play

jwt := io.jwt.encode_sign(
	{"alg": "HS256"},
	{"ä, ü, ï, ö, ë, ÿ": "ä, ü, ï, ö, ë, ÿ"},
	{"kty": "oct", "k": "Zm9v"},
)

dec := io.jwt.decode(jwt)
ver := io.jwt.verify_hs256(jwt, "foo")
```

**Comment by JohnHoward9** [comment]
Thanks everyone for their response.  We are still seeing the issue, but have now ruled out non-latin characters as we saw the issue yesterday with a JWT that did not have non latin characters.  
Unfortunately, we are only seeing this in prod and we only get one or two occurrences per day, so difficult to debug.  However, we have added more logging to see it we can get to the root cause.  I'll close this ticket
